portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -39,27 +40,42 @@
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    String defaultCommentLimit = "5";
+    int commentLimit = Integer.parseInt(getParameter(request, "numShown", defaultCommentLimit));
+
+    String defaultPageNumber = "0";
+    int pageNumber = Integer.parseInt(getParameter(request, "pageNumber", defaultPageNumber));
 
+    String defaultSortMethod = "DESCENDING";
+    String sortMethod = getParameter(request, "sortMethod", defaultSortMethod);
+
+    Query query = new Query("Comment");
+    
+    if (sortMethod.equals(defaultSortMethod)) {
+      query.addSort("timestamp", SortDirection.DESCENDING);
+    } else {
+      query.addSort("timestamp", SortDirection.ASCENDING);
+    }
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
+    
 
     ArrayList<Comment> comments = new ArrayList<>();
-    String defaultCommentLimit = "5";
-    int commentLimit = Integer.parseInt(getParameter(request, "numShown",
-                                                    "5"));
+
+    int startingIndex = pageNumber * commentLimit;
     int count = 0;
 
     for (Entity entity : results.asIterable()) {
-      long id = entity.getKey().getId();
-      String comment = (String) entity.getProperty("comment");
-      long timestamp = (long) entity.getProperty("timestamp");
-
-      Comment newComment = new Comment(id, comment, timestamp);
-      comments.add(newComment);
+      if (count >= startingIndex) {
+        long id = entity.getKey().getId();
+        String comment = (String) entity.getProperty("comment");
+        long timestamp = (long) entity.getProperty("timestamp");
 
+        Comment newComment = new Comment(id, comment, timestamp);
+        comments.add(newComment);
+      }
       count++;
-      if(count == commentLimit) {
+      if(count == startingIndex + commentLimit) {
```

Comment:
You can just do comments.size() == commentLimit it's the same thing but more straightforward